gpage.h

```diff
@@ -222,16 +235,19 @@ namespace gcpp {
 	//
 	inline
 	bool gpage::contains(gsl::not_null<const byte*> p) const noexcept {
-		return &storage[0] <= p && p < &storage[total_size - 1];
+		//  Use std::less<> to compare (possibly unrelated) pointers portably
+		auto const cmp = std::less<>{};
+		auto const ext = extent();
+		return !cmp(p, ext.data()) && cmp(p, ext.data() + ext.size());
```

Comment:
The standard seems to leave unspecified what happens when using `<` to compare pointers to different 'objects'. It also doesn't specify what total order is required for `std::less` and friends. So, I'm curious, is `std::less` required to implement a total order which results in this working?

I believe the weakest additional requirement on `std::less` that would be needed for it to be obviously useful in this case, is:

For all pointers `p` and `q`, for which `p < q` is specified, `std::less()(p, q)` is required to return the same value.

Given that you can address every byte (and bit) of an object (including padding) through `char*`, there would be no way to implement a total order which obeys this property without also implementing one which allows detecting whether a pointer is within a given allocation.

@hsutter What do you think? Does the current standard provide sufficient guarantees for this to work?